File: app-server/src/main/java/com/github/arhor/simple/expense/tracker/service/mapping/IgnoreAuditProps.java
package com.github.arhor.simple.expense.tracker.service.mapping;

import java.lang.annotation.ElementType;
import java.lang.annotation.Retention;
import java.lang.annotation.RetentionPolicy;
import java.lang.annotation.Target;

import org.mapstruct.Mapping;

@Retention(RetentionPolicy.RUNTIME)
@Target(ElementType.METHOD)
@Mapping(target = "created", ignore = true)
@Mapping(target = "updated", ignore = true)
public @interface IgnoreAuditProps {
}

File: app-server/src/main/kotlin/com/github/arhor/simple/expense/tracker/service/mapping/IgnoreAuditProps.kt
package com.github.arhor.simple.expense.tracker.service.mapping

import org.mapstruct.Mapping

@Retention(AnnotationRetention.RUNTIME)
@Target(AnnotationTarget.FUNCTION)
@Mapping(target = "created", ignore = true)
@Mapping(target = "updated", ignore = true)
annotation class IgnoreAuditProps
